app/settings/forms.py

```diff
@@ -13,14 +19,29 @@ class PreferencesForm(FlaskForm):
     submit = SubmitField('Save Changes')
 
     def validate_username(self, username):
-        # Need to validate that username is not already used.
-        pass
+        """ Checks if the data has changed, if it has then run validation from app/auth/RegistrationForm"""
+        if current_user.username.lower() != username.data.lower():
+            print(f'current:{current_user.username} new: {username.data}')
+            authenticator.validate_username(self, username)
 
+    
     def validate_email(self, email):
-        # Need to validate that email is not already used.
-        pass
+        """ Checks if the data has changed, if it has then run validation from app/auth/RegistrationForm"""
+        if current_user.email.lower() != email.data.lower():
+            print(f'current:{current_user.email} new: {email}')
+            authenticator.validate_email(self, email)
 
+    def validate_phone_number(self, phone_number):
+        """ Checks if the data has changed, if it has then first strip valid characters, 
```

Comment:
Can you put in the documentation what the parameter "phone_number" do?

Reply:
Done.